src/wallet/walletdb.cpp

```diff
@@ -651,6 +728,32 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
 
             // ignore earliest key creation time as taddr will exist before any zaddr
         }
+        else if (strType == "sapzkeymeta")
+        {
+            libzcash::SaplingIncomingViewingKey ivk;
+            ssKey >> ivk;
+            CKeyMetadata keyMeta;
+            ssValue >> keyMeta;
+
+            wss.nZKeyMeta++;
+
+            pwallet->LoadSaplingZKeyMetadata(ivk, keyMeta);
```

Comment:
Currently `LoadSaplingZKeyMetadata` always returns `true` as noted by @bitcartel, but it does return a boolean, so this is a maintenance hazard. Please file a ticket.

Reply:
Opened #3571.